provider/core/src/resource.rs

```diff
@@ -264,26 +264,29 @@ fn test_path_syntax() {
     // No version:
     assert_eq!(
         ResourceKey::construct_internal(tagged!("hello/world")),
-        Err(("[a-zA-z0-9_/@]", 25))
+        Err((
+            "[a-zA-z0-9_/@]",
+            concat!(leading_tag!(), "hello/world").len()
+        ))
     );
 
     assert_eq!(
         ResourceKey::construct_internal(tagged!("hello/world@")),
-        Err(("[0-9]", 26))
+        Err(("[0-9]", concat!(leading_tag!(), "hello/world@").len()))
```

Comment:
You can go back to the NUL thing if you want

Reply:
Actually I still am more comfy with having something besides just a NUL deliminating the strings. We are introducing these identifiers into the lexicon, and it is not at all unreasonable to think that they might end up in legitimate programs. We should have *some* ICU4X-specific prefix and/or suffix to label them.